### PR TITLE
Add logger for Azure to see usage of scheduler/job_queue

### LINKED ISSUE #7162: Make scheduler logging explicit
Create an explicit logging entry (Azure-logger) indicating usage of scheduler and job queue, and which queue system was used.

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #7162 

**Approach**
This PR adds a logger specifically for Azure, to see the usage of scheduler vs job_queue and which queue system it is being used with.

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/ensemble_evaluator/_builder/_legacy.py [hunk 1]
```diff
@@ -39,6 +39,7 @@
 
 logger = logging.getLogger(__name__)
 event_logger = logging.getLogger("ert.event_log")
+job_queue_or_scheduler_logger = logging.getLogger("ert.job_queue_or_scheduler_log")
```
Comment (xjules) [inline src/ert/ensemble_evaluator/_builder/_legacy.py:42]: I still don't like the name tbh. What about just scheduler_logger?

Comment (jonathan-eq) [reply src/ert/ensemble_evaluator/_builder/_legacy.py:42]: We already have a ert.scheduler logger. Do you think I could reuse that one?

Comment (jonathan-eq) [reply src/ert/ensemble_evaluator/_builder/_legacy.py:42]: I interpreted the issue as "make a specific logger only for this purpose", but I do not think it matters that we re-use the scheduler one.

**Review by xjules — APPROVED** [review]
Nice job Jonathan! :rocket:

**Comment by jonathan-eq** [comment]
The commit in this PR will add this to the main ert log, which should be retrievable in Azure. (only the ert.scheduler part is new)

```
2024-02-14 09:01:18,119 - ert.callbacks - LegacyEnsemble - INFO - Loaded POLY_RES
2024-02-14 09:01:18,121 - ert.callbacks - LegacyEnsemble - INFO - Saved POLY_RES to storage
2024-02-14 09:01:18,124 - ert.scheduler - LegacyEnsemble - INFO - Experiment ran on QUEUESYSTEM: LOCAL
2024-02-14 09:01:18,125 - ert.scheduler - LegacyEnsemble - INFO - Experiment ran on ORCHESTRATOR: scheduler
2024-02-14 09:01:20,041 - ert.gui.simulation.tracker_worker - Dummy-2 - DEBUG - emit SnapshotUpdateEvent(phase_name='Running forecast for iteration: 0', current_phase=0, total_phases=1, progress=1.0, indeterminate=False, iteration=0, partial_snapshot=<ert.ensemble_evaluator.snapshot.PartialSnapshot object at 0x1869d79d0>)

```

**Comment by jonathan-eq** [comment]
Running without scheduler logs this:
```
2024-02-14 09:08:10,958 - ert.job_queue - Thread-100 (_job_monitor) - DEBUG - Set "poly_96"(96) to JOB_QUEUE_SUCCESS
2024-02-14 09:08:11,395 - ert.gui.simulation.tracker_worker - Dummy-2 - DEBUG - emit SnapshotUpdateEvent(phase_name='Running forecast for iteration: 0', current_phase=0, total_phases=1, progress=0.53, indeterminate=False, iteration=0, partial_snapshot=<ert.ensemble_evaluator.snapshot.PartialSnapshot object at 0x1819052d0>)
2024-02-14 09:08:11,697 - ert.scheduler - LegacyEnsemble - INFO - Experiment ran on QUEUESYSTEM: LOCAL
2024-02-14 09:08:11,697 - ert.scheduler - LegacyEnsemble - INFO - Experiment ran on ORCHESTRATOR: job_queue
2024-02-14 09:08:13,415 - ert.gui.simulation.tracker_worker - Dummy-2 - DEBUG - emit SnapshotUpdateEvent(phase_name='Running forecast for iteration: 0', current_phase=0, total_phases=1, progress=1.0, indeterminate=False, iteration=0, partial_snapshot=<ert.ensemble_evaluator.snapshot.PartialSnapshot object at 0x18321c130>)

```

**Pull request merged**